diff --git a/godotsteam_multiplayer_peer.cpp b/godotsteam_multiplayer_peer.cpp
index 91ff927..de94799 100644
--- a/godotsteam_multiplayer_peer.cpp
+++ b/godotsteam_multiplayer_peer.cpp
@@ -360,7 +360,7 @@ void SteamMultiplayerPeer::_close() {
 	connection_status = CONNECTION_DISCONNECTED;
 	server = false;
 
-	if (Engine::get_singleton()->has_singleton("Steam")) {
+	if (!Engine::get_singleton()->has_singleton("Steam")) {
 		return;
 	}
 
